TLM/TLM/Util/TrackUtils.cs

```diff
@@ -20,11 +20,28 @@ internal static class TrackUtils {
                 !laneInfo.m_vehicleType.IsFlagSet(~VEHICLE_TYPES);
         }
 
-        public static bool IsBidirectional(this NetInfo.Lane laneInfo) {
-            // Note that for station tracks we have:
-            // AvoidForward = Both + 4
-            // AvoidBackward = Both + 8
-            return laneInfo.m_direction.CheckFlags(NetInfo.Direction.Both);
-        }
+        /// <summary>
+        /// checks if vehicles move backward or bypass backward (considers LHT)
+        /// </summary>
+        /// <returns>true if vehicles move backward including AvoidForward,
+        /// false if vehicles going forward, bi-directional, or non-directional</returns>
+        internal static bool IsGoingBackward(this NetInfo.Direction direction) =>
+            (direction & NetInfo.Direction.Both) == NetInfo.Direction.Backward ||
+            (direction & NetInfo.Direction.AvoidBoth) == NetInfo.Direction.AvoidForward;
+
+        internal static bool IsGoingForward(this NetInfo.Direction direction) =>
+            (direction & NetInfo.Direction.Both) == NetInfo.Direction.Forward ||
+            (direction & NetInfo.Direction.AvoidBoth) == NetInfo.Direction.AvoidBackward;
+
+        /// <summary>
+        /// checks if vehicles move backward or bypass backward (considers LHT)
+        /// </summary>
+        /// <returns>true if vehicles move backward including AvoidForward,
+        /// false if vehicles going ward, bi-directional, or non-directional</returns>
+        internal static bool IsGoingBackward(this NetInfo.Lane laneInfo) =>
+            laneInfo.m_finalDirection.IsGoingBackward();
+
+        internal static bool IsGoingForward(this NetInfo.Lane laneInfo) =>
+            laneInfo.m_finalDirection.IsGoingForward();
```

Comment:
Would it make more sense to put these in `NetLaneExtensions.cs` ?

While unlikely, there's nothing to stop asset creators from adding the `Avoid` directions to any net lane on any type of network? So this approach to checking direction should be used as standard IMO.

Also, why no xmldoc for the `IsGoingForward` extensions?

Reply:
This is `NetInfo.Lane` though not `NetLane`. I can create a new class. 

> While unlikely, there's nothing to stop asset creators from adding the Avoid directions to any net lane on any type of network?

Defiantly not unlikely. even vanila assets use `AvoidForward`/`AvoidBackward` and I have checked for them here.
Unless if you mean the `Avoid` direction. It would make a the lane symmetrical in both directions (bidirectional) and as the doc suggest I am to return false.

> Also, why no xmldoc for the IsGoingForward extensions?

forgot.